lib/base-compiler.ts

```diff
@@ -1608,16 +1607,7 @@ export class BaseCompiler implements ICompiler {
     async handleExecution(key, executeParameters): Promise<CompilationResult> {
         if (this.compiler.interpreted) return this.handleInterpreting(key, executeParameters);
         const buildResult = await this.getOrBuildExecutable(key);
-        if (buildResult.code !== 0) {
-            return {
-                code: -1,
-                didExecute: false,
-                buildResult,
-                stderr: [{text: 'Build failed'}],
-                stdout: [],
-                timedOut: false,
-            };
-        } else {
+        if (buildResult.code === 0) {
```

Comment:
I think this should probably be turned into an early exit situation instead of having an `else`.